timefold-solver-python-core/src/main/python/score/_score_analysis.py

```diff
@@ -445,14 +466,21 @@ class ConstraintAnalysis(Generic[Score_]):
          but still non-zero constraint weight; non-empty if constraint has matches.
          This is a list to simplify access to individual elements,
          but it contains no duplicates just like `set` wouldn't.
-
+    summary : str
+        Returns a diagnostic text
+        that explains part of the score quality through the ConstraintAnalysis API.
+    match_count : int
+        Return the match count of the constraint.
     """
     _delegate: '_JavaConstraintAnalysis[Score_]'
 
     def __init__(self, delegate: '_JavaConstraintAnalysis[Score_]'):
         self._delegate = delegate
         delegate.constraintRef()
 
+    def __str__(self):
+        return self.summarize
```

Comment:
I see the point of turning `summarize` into `summary` here, but I think it's a misunderstanding of the concept.

The point of calling the method `summarize` is to indicate that there is an active process behind it. It's not an attribute - it is a method which has to do a lot of processing to product a result. And it will do it again the next time it is called. It is not a value, it is an actual function.

I believe treating this as a method is the correct thing to do. (And coincidentally, it would be consistent with the Java API.)

Reply:
I'm okay with both approaches, but I've argued that `summarize` is a method, and we should keep it consistent with the Java API.

@Christopher-Chianelli, are you okay with reverting it to `summarize`?